### ISSUE TITLE
a text objects works incorrectly in nested objects

### ISSUE THREAD

**Issue by asvetliakov** [VSCodeVim/Vim#1049]
Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----

### What did you do?
```ts
const a = {
  b: {
    /*cursor here*/c: "d"
  }
}
```

Put the exact keys you pressed.
```ca{```

### What did you expect to happen?
```ts
const a = {
  b: /* cursor here for ca{ */
}
```
### What happened instead?
```ts
const a = {
  b: /* cursor */
  }
}
```
### Technical details:

* VSCode Version: 1.7.1
* VsCodeVim Version: *0.4.1*
* OS: Windows 10

same happens with ```ca[ , ca}, ca]``` and probably others

**Comment by asvetliakov** [comment]
seems related to #1023

**Comment by xconverge** [comment]
@rebornix this should be fixed as well now that #1023 is fixed

**Comment by rebornix** [comment]
Thanks @xconverge !
